src/main/java/org/jenkinsci/plugins/workflow/steps/StepDescriptor.java

```diff
@@ -273,7 +273,20 @@ public final void checkContextAvailability(StepContext c) throws MissingContextV
             @SuppressFBWarnings(value="NP_PARAMETER_MUST_BE_NONNULL_BUT_MARKED_AS_NULLABLE", justification="all() will not return nulls")
             @Override
             public boolean apply(StepDescriptor i) {
-                return i.isMetaStep();
+                if (i.isMetaStep()) {
+                    Class<?> a = i.getMetaStepArgumentType();
+                    if (a != null) {
```

Comment:
Can skip `isMetaStep` call here in such a case, since `getMetaStepArgumentType` already covers that.

Reply:
Gotcha.